sample_app/app_utils.c

```diff
@@ -0,0 +1,714 @@
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2021 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#define _GNU_SOURCE /* For asprintf() */
+
+#include "app_utils.h"
+#include "app_log.h"
+#include "app_gsdml.h"
+#include "sampleapp_common.h"
+#include "osal.h"
+#include "osal_log.h" /* For LOG_LEVEL */
+#include "pnal.h"
+#include <pnet_api.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#define GET_HIGH_BYTE(id) ((id >> 8) & 0xFF)
+#define GET_LOW_BYTE(id)  (id & 0xFF)
+
+void app_utils_ip_to_string (pnal_ipaddr_t ip, char * outputstring)
+{
+   snprintf (
+      outputstring,
+      PNAL_INET_ADDRSTR_SIZE,
+      "%u.%u.%u.%u",
+      (uint8_t) ((ip >> 24) & 0xFF),
+      (uint8_t) ((ip >> 16) & 0xFF),
+      (uint8_t) ((ip >> 8) & 0xFF),
+      (uint8_t) (ip & 0xFF));
+}
+
+void app_utils_mac_to_string (pnet_ethaddr_t mac, char * outputstring)
+{
+   snprintf (
+      outputstring,
+      PNAL_ETH_ADDRSTR_SIZE,
+      "%02X:%02X:%02X:%02X:%02X:%02X",
+      mac.addr[0],
+      mac.addr[1],
+      mac.addr[2],
+      mac.addr[3],
+      mac.addr[4],
+      mac.addr[5]);
+}
+
+const char * app_utils_submod_dir_to_string (pnet_submodule_dir_t direction)
+{
+   const char * s = "<error>";
+
+   switch (direction)
+   {
+   case PNET_DIR_NO_IO:
+      s = "NO_IO";
+      break;
+   case PNET_DIR_INPUT:
+      s = "INPUT";
+      break;
+   case PNET_DIR_OUTPUT:
+      s = "OUTPUT";
+      break;
+   case PNET_DIR_IO:
+      s = "INPUT_OUTPUT";
+      break;
+   }
+
+   return s;
+}
+
+const char * app_utils_ioxs_to_string (pnet_ioxs_values_t ioxs)
+{
+   const char * s = "<error>";
+   switch (ioxs)
+   {
+   case PNET_IOXS_BAD:
+      s = "IOXS_BAD";
+      break;
+   case PNET_IOXS_GOOD:
+      s = "IOXS_GOOD";
+      break;
+   }
+
+   return s;
+}
+
+void app_utils_get_error_code_strings (
```

Comment:
Great!

Reply:
We should add more error codes but thats for another commit